### PR TITLE
Cni/migrate endpoint get rolling stock

### LINKED ISSUE #3180: Migrate endpoint GET /rollingstock/ to editoast
# Description
Migrate `GET /rolling-stock/` endpoint to editoast.

# Goal
Drop API service.

# Acceptance Criteria
- [ ] the frontend uses the editoast endpoint and the user does not see any difference
- [ ] the endpoint doesn't exist anymore in API

### PR DESCRIPTION
**Pull request by clarani**
Closes #3180

### CODE REVIEW

File: editoast/src/views/electrical_profiles.rs [hunk 1]
```diff
@@ -120,7 +120,7 @@ impl ElectricalProfileSet {
 #[derive(Debug, Error, EditoastError)]
 #[editoast_error(base_id = "electrical_profiles", context = "Self::context")]
 pub enum ElectricalProfilesError {
-    /// Couldn't found the infra with the given id
+    /// Couldn't found the electrical profile with the given id
```
Comment (Castavo) [inline editoast/src/views/electrical_profiles.rs]: Might as well :grin: 
```suggestion
    /// Couldn't find the electrical profile set with the given id
```

File: editoast/src/views/rolling_stocks.rs [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+mod tests {
+    use super::RollingStock;
+    use crate::client::PostgresConfig;
+    use crate::tables::osrd_infra_rollingstock::dsl;
+    use actix_http::StatusCode;
+    use diesel::prelude::*;
+    use diesel::result::Error;
+    use serde_json::json;
+
+    fn test_ep_set_transaction(fn_test: fn(&mut PgConnection, RollingStock)) {
```
Comment (Castavo) [inline editoast/src/views/rolling_stocks.rs]: You have forgotten to change the test name

**Review by Castavo — DISMISSED** [review]
Nice PR !

File: editoast/src/views/rolling_stocks.rs (hunk 2 above)
Comment (clarani) [reply editoast/src/views/rolling_stocks.rs]: Indeed ! Thanks :)

File: editoast/src/views/rolling_stocks.rs [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    pub startup_acceleration: f64,
+    pub comfort_acceleration: f64,
+    pub gamma: JsonValue,
+    pub inertia_coefficient: f64,
+    pub features: Vec<String>,
+    pub mass: f64,
+    pub rolling_resistance: JsonValue,
+    pub loading_gauge: String,
+    pub metadata: JsonValue,
+}
```
Comment (flomonster) [inline editoast/src/views/rolling_stocks.rs]: We can remove all these `pub`

File: editoast/src/views/rolling_stocks.rs [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+/// Return a specific set of electrical profiles
+#[get("/{rolling_stock}")]
+async fn get(db_pool: Data<DbPool>, rolling_stock: Path<i64>) -> Result<Json<RollingStock>> {
+    let rolling_stock = rolling_stock.into_inner();
+    block(move || {
+        let mut conn = db_pool.get().expect("Failed to get DB connection");
+        Ok(Json(RollingStock::retrieve(&mut conn, rolling_stock)?))
+    })
+    .await
+    .unwrap()
```
Comment (flomonster) [inline editoast/src/views/rolling_stocks.rs]: We should block and get a connection as late as possible.
The idea is to use `db_pool` everywhere and when you need to make a request, you block and get a connection.

You will need a `DbPool` for your tests. You can take example here: https://github.com/DGEXSolutions/osrd/blob/ad34f7ce580b1f150abada912f6344ee7450cac2/editoast/src/documents.rs#L154-L155

I know that's not what we're currently doing in most endpoints, but that should be fixed.

File: editoast/src/views/rolling_stocks.rs [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    use diesel::prelude::*;
+    use diesel::result::Error;
+    use serde_json::json;
+
+    fn test_rolling_stock_set_transaction(fn_test: fn(&mut PgConnection, RollingStock)) {
+        let mut conn = PgConnection::establish(&PostgresConfig::default().url()).unwrap();
+        conn.test_transaction::<_, Error, _>(|conn| {
+            diesel::delete(dsl::osrd_infra_rollingstock).execute(conn)?;
+
+            let rolling_stock = RollingStock {
```
Comment (flomonster) [inline editoast/src/views/rolling_stocks.rs]: We could have an example of rolling stock in a test directory and load it from the test.

**Review by flomonster — CHANGES_REQUESTED** [review]
Thanks, for your PR. You forgot to update openapis (editoast + API)

File: editoast/src/schema/rolling_stock.rs [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    use diesel::PgConnection;
+    use std::fs;
+
+    #[actix_test]
+    async fn create_get_delete_rolling_stock() {
+        let manager = ConnectionManager::<PgConnection>::new(PostgresConfig::default().url());
+        let db_pool = Data::new(Pool::builder().max_size(2).build(manager).unwrap());
+
+        let path = "./src/tests/example_rolling_stock.json";
+        let data = fs::read_to_string(path).expect("Unable to read file");
```
Comment (flomonster) [inline editoast/src/schema/rolling_stock.rs]: You can use `include_str!` instead.

File: editoast/src/views/rolling_stocks.rs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+use actix_web::get;
+use actix_web::web::{self, block, Data, Json, Path};
+
+pub fn routes() -> impl HttpServiceFactory {
+    web::scope("/rolling_stock").service(get)
+}
+
+/// =================================
+/// Endpoints
+/// =================================
```
Comment (flomonster) [inline editoast/src/views/rolling_stocks.rs]: This is not a good idea. I think you take this from `electrical_profiles.rs`. It's not a good idea. It's actually doing something weird on the documentation.

**Review by flomonster — CHANGES_REQUESTED** [review]
Almost there. Can you also change `python/api/openapi.yaml` to mark the endpoint as deprecated.

File: editoast/src/schema/rolling_stock.rs [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+        match self {
+            RollingStockError::NotFound(rolling_stock_id) => json!({
+                "rolling_stock_id": rolling_stock_id,
+            })
+            .as_object()
+            .cloned()
+            .unwrap(),
+        }
+    }
+}
```
Comment (flomonster) [inline editoast/src/schema/rolling_stock.rs]: You don't need this anymore checkout this PR: https://github.com/DGEXSolutions/osrd/pull/3389

TLDR: You just have to adapt your enum with named variant:

```rust
[derive(Debug, Error, EditoastError)]
#[editoast_error(base_id = "rollingstocks")]
enum RollingStockError {
    #[error("Rolling stock '{rolling_stock_id}', could not be found")]
    #[editoast_error(status = 404)]
    NotFound{rolling_stock_id: i64},
}
```

File: editoast/src/views/rolling_stocks.rs [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+pub fn routes() -> impl HttpServiceFactory {
+    web::scope("/rolling_stock").service(get)
+}
+
+/// Return a specific set of electrical profiles
+#[get("/{rolling_stock}")]
+async fn get(db_pool: Data<DbPool>, rolling_stock: Path<i64>) -> Result<Json<RollingStock>> {
+    let rolling_stock = rolling_stock.into_inner();
+    block(move || Ok(Json(RollingStock::retrieve(db_pool, rolling_stock)?)))
```
Comment (flomonster) [inline editoast/src/views/rolling_stocks.rs]: The block should be done in the `RollingStock::retrieve` function just before you take a DB connection.

**Review by flomonster — CHANGES_REQUESTED** [review]

File: editoast/src/schema/rolling_stock.rs [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+                Err(e) => Err(e.into()),
+            }
+        })
+        .await
+        .unwrap()
+    }
+
+    pub async fn delete(db_pool: Data<DbPool>, rolling_stock_id: i64) -> Result<()> {
+        block::<_, Result<_>>(move || {
+            use crate::tables::osrd_infra_rollingstock::dsl::*;
```
Comment (Castavo) [inline editoast/src/schema/rolling_stock.rs:98]: why not use `dsl::osrd_infra_rollingstock` like in `create()` ?

File: editoast/src/views/rolling_stocks.rs [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    use actix_http::StatusCode;
+    use actix_web::test as actix_test;
+    use actix_web::test::{call_service, TestRequest};
+    use actix_web::web::Data;
+    use diesel::r2d2::{ConnectionManager, Pool};
+    use diesel::PgConnection;
+    use std::fs;
+
+    #[actix_test]
+    async fn get_rolling_stock() {
```
Comment (Castavo) [inline editoast/src/views/rolling_stocks.rs:36]: Nice test :grin:

File: editoast/src/schema/rolling_stock.rs (hunk 10 above)
Comment (clarani) [reply editoast/src/schema/rolling_stock.rs:98]: Hum I tried to remove it, but it doesn't work without this line :/ So I would rather keep it this way

**Review by flomonster — APPROVED** [review]
